Source/model/draw/ColorMatrixEffect.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+
+open class ColorMatrixEffect: Effect {
+
+    public let matrix: [Double]
+
+    public init(matrix: [Double] = [], input: Effect? = nil) {
+        self.matrix = matrix
+        super.init(input: input)
+    }
+
+    public init(color: Color, input: Effect? = nil) {
+        self.matrix = [Double(color.r()) / 255.0, 0, 0, 0, 0,
+                       0, Double(color.g()) / 255.0, 0, 0, 0,
+                       0, 0, Double(color.b()) / 255.0, 0, 0,
+                       0, 0, 0, Double(color.a()) / 255.0, 0]
+        super.init(input: input)
+    }
+
+    public init(saturate: Double, input: Effect? = nil) {
+        let s = max(min(saturate, 1), 0)
+        self.matrix = [0.213 + 0.787 * s, 0.715 - 0.715 * s, 0.072 - 0.072 * s, 0, 0,
+                       0.213 - 0.213 * s, 0.715 + 0.285 * s, 0.072 - 0.072 * s, 0, 0,
+                       0.213 - 0.213 * s, 0.715 - 0.715 * s, 0.072 + 0.928 * s, 0, 0,
+                       0, 0, 0, 1, 0]
+        super.init(input: input)
+    }
+
+    public init(hueRotate: Double, input: Effect? = nil) {
```

Comment:
There is a good practice: all angles in radians. When SVG using degrees, convert it in the parser.